source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,41 @@
 class UrlsController < ApplicationController
+
+
+	def index
+		@urls= Url.all
+	end
+
+  def create
+    @url = Url.new(url_params)
+
+    if !@url.save
+     flash[:error] = "Invalid url. Please Enter a valid url"
+    end
+
+    redirect_to urls_path
+  end
+
+	def new
+			@url = UrlsControllerrl.new
```

Comment:
What's a "UrlsControllerrl"? :)

Reply:
Ohh. Sorry,
1. I missed removing the new function
2. I guess that missed type for URL.new

Will change it sorry :P